src/serverenvironment.cpp

```diff
@@ -2049,18 +2064,26 @@ void ServerEnvironment::activateObjects(MapBlock *block, u32 dtime_s)
 		if (!obj) {
 			errorstream << "ServerEnvironment::activateObjects(): "
 				<< "failed to create active object from static object "
-				<< "in block " << (s_obj.pos / BS)
+				<< "in block " << block->getPos()
 				<< " type=" << (int)s_obj.type << " data:" << std::endl;
 			print_hexdump(verbosestream, s_obj.data);
 
 			new_stored.push_back(s_obj);
 			continue;
 		}
-		verbosestream << "ServerEnvironment::activateObjects(): "
-			<< "activated static object pos=" << (s_obj.pos / BS)
-			<< " type=" << (int)s_obj.type << std::endl;
+
+		obj->m_static_exists = true;
+		obj->m_static_block = block->getPos();
+
 		// This will also add the object to the active static list
-		addActiveObjectRaw(std::move(obj), false, dtime_s);
+		bool ok = addActiveObjectRaw(std::move(obj), &s_obj, dtime_s) != 0;
```

Comment:
you can assign & check in the condition directly

Reply:
that's possible but I don't see the advantage